src/components/ModalWithConfirm.js

```diff
@@ -0,0 +1,30 @@
+import Modal from "./Modal";
+
+export default class ModalWithConfirm extends Modal {
+  constructor(modalSelector) {
+    super({modalSelector});
+    this._form = this._modalElement.querySelector(".modal__form");
+    this._modalButton = this._modalElement.querySelector(".modal__button");
+  }
+
+  handleDelete(handleDeleteSubmit) {
+    this._handleDeleteSubmit = handleDeleteSubmit;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleDeleteSubmit();
+    });
+  }
+
+  renderLoading() {
+    if (!this._modalButton) {
+        console.error('Element not found');
+        return;
+    }
+    this._modalButton .textContent = "Loading...";
```

Comment:
Needs correction: you should change the text to `Loading...` if loading otherwise you should reset the text to the initial state, like you do that for other forms.